cidr-filtering-cni

```diff
@@ -0,0 +1,252 @@
+        else
+            ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${table_type} ${NFT_TABLE} m-${policy_id}-${direction}-ports ${match_iface} $CNI_IFNAME ${protocol,,} dport ${port} counter meta mark set meta mark "|" 0x00010000)
+        fi
+    }
+
+    # handle ip block
+    if [[ $(echo "$stdin" | jq -r ".${direction}.subnets") != "null" ]]; then
+        for_json_array "$stdin" ".${direction}.subnets" _create_subnet_rule
+    fi
+
+    # handle ports
+    if [[ $(echo "$stdin" | jq -r ".${direction}.ports") != "null" ]]; then
+        for_json_array "$stdin" ".${direction}.ports" _accept_port
+    fi
+}
+
+exitWithError() {
+    echo "{"cniVersion": "$(jq -r ".cniVersion" <<< "$stdin")","msg":"$1","code":1,"details":"$2"}" >&3
+    exit 1
+}
+
+exitWithSuccess() {
+    prev_result=$(get_object "${stdin}" ".prevResult")
+    if [[ "$prev_result" == "null" ]]; then
+        echo "{"cniVersion": "$(jq -r ".cniVersion" <<< "$stdin")"}" >&3
+    else
+        echo "$prev_result" >&3
+    fi
+    exit 0
+}
+
+create_chains_for_filtering() {
+    local policy_id="$1"
+    local table_type="$2"
+    local direction="$3"
+
+    local match_iface="iifname"
+    if [[ "$direction" == "egress" ]]; then
+        local match_iface="oifname"
+    fi
+
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}") || exitWithError "Failed to add chain m-${policy_id}-${direction}"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${direction}" "${match_iface}" "$CNI_IFNAME" counter jump m-"${policy_id}"-"${direction}") || exitWithError "Failed to add rule: jump m-${policy_id}-${direction}"
+
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}"-subnets) || exitWithError "Failed to add chain m-"${policy_id}"-"${direction}"-subnets"
+    ip netns exec "$CNI_CONTAINERID" $(create_chain "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}"-ports) || exitWithError "Failed to add chain m-"${policy_id}"-"${direction}"-ports"
+    
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}" counter jump m-"${policy_id}"-"${direction}"-subnets) || exitWithError "Failed to add rule: jump m-"${policy_id}"-"${direction}"-subnets"
+    ip netns exec "$CNI_CONTAINERID" $(nft_add_rule "${table_type}" "${NFT_TABLE}" m-"${policy_id}"-"${direction}" counter jump m-"${policy_id}"-"${direction}"-ports) || exitWithError "Failed to add rule: jump m-"${policy_id}"-"${direction}"-ports"
+
+    create_rules_for_filtering "${policy_id}" "${table_type}" "${direction}" "${match_iface}"
+}
+
+case $CNI_COMMAND in
+ADD)
+    echo "CNI_NETNS: $CNI_NETNS" >> $logFile
+    echo "CNI_CONTAINERID: $CNI_CONTAINERID" >> $logFile
+    echo "STDIN: $stdin" >> $logFile
+
+    policy_name=$(echo "$stdin" | jq -r ".name")
+    policy_id="$(echo "$policy_name" | sha1sum )"
```

Comment:
Does it make sense to use a different algorithm here ? A symmetric one, so you can also go from `policy_id` to `policy_name` .

Reply:
the name length of a nft chain is limited to 32 chars, so even with symmetric algorithm, I would need to strip the result.
So I wouldn't be able to get to the original string anyway.

Reply:
encryption algorithm would also likely yield unprintable characters, that would need to be dealt with.
If we want to make it easier for use, it would be maybe better to not do this at all and just strip the original name to the desired length. If two or more configurations happen to have the same name, it would still work we would just end up with all the rules in one chain.